src/engine/naive_engine.cc

```diff
@@ -118,6 +179,15 @@ class NaiveEngine final : public Engine {
   // callback to oncomplete
   static void OnComplete(Engine *engine, void *param) {
     static_cast<NaiveEngine*>(engine)->req_completed_ = true;
+#if MXNET_USE_PROFILER
+    if (param != nullptr) {
```

Comment:
This is not necessary. You can do this in push and pushasync directly since everything is sync